### ISSUE TITLE
[FEATURE] Improve XGBoost model accuracy with better features, tuning & validation

### ISSUE THREAD

**Issue by aryanmalhotraug25-hash** [Arnav-Singh-5080/IPL-Win-Probability-Prediction#328]
## Description
The current prediction pipeline already uses **XGBoost** with `ColumnTransformer` + `OneHotEncoder` inside a `Pipeline`, achieving ~81% precision and ~83% recall. We can push accuracy further with better feature engineering, tuning, and validation strategy.

## Current Setup
- **Model:** XGBoost Classifier  
- **Preprocessing:** ColumnTransformer + OneHotEncoder  
- **CV Strategy:** 5-Fold Stratified K-Fold  
- **Precision:** 81.61% | **Recall:** 83.52%

> Note: The Model Performance UI currently labels the active classifier as "Logistic Regression" — this should also be corrected to reflect the actual XGBoost model.

<img width="1885" height="904" alt="Image" src="https://github.com/user-attachments/assets/fb31ad31-45e5-45e5-a1d4-49a2ee97ce22" />

### 1. 🧠 Better Feature Engineering
- Run rate differential (`RRR - CRR`)
- Wickets remaining & balls remaining
- Last 5 overs momentum (runs & wickets)
- Pressure index (wickets lost × required rate)
- Venue-specific avg first-innings score
- Head-to-head team win ratio
- Toss winner & toss decision flag

### 2. ⚙️ Hyperparameter Tuning with Optuna
Tune XGBoost parameters like `n_estimators`, `max_depth`, `learning_rate`, `subsample`, `colsample_bytree`, `reg_alpha`, `reg_lambda`.

### 3. 📊 Smarter Cross-Validation
Replace Stratified K-Fold with **GroupKFold grouped by `match_id`** to prevent leakage (same match in train + test).

### 4. 🔤 Better Encoding
Replace OneHotEncoder for high-cardinality columns (teams, venues, cities) with **Target Encoding** to reduce dimensionality.

### 5. 🤝 Model Ensembling
Stack XGBoost with LightGBM & CatBoost for marginal but consistent gains.

## Expected Outcome
Push model accuracy from **~81% → 88–92%** with more reliable, leak-free predictions.

## Checklist
- [x] Add new engineered features
- [x] Switch to GroupKFold CV
- [ ] Hyperparameter tuning with Optuna
- [x] Apply target encoding for high-cardinality features
- [x] Retrain & update `pipe.pkl`
- [x] Fix UI label from "Logistic Regression" → "XGBoost"

---
## Hi @Arnav-Singh-5080 , Please assign this under GSSoC'26

**Comment by iyashsharmahere** [comment]
Can you assign this issue to me
@iyashsharmahere

**Comment by Vectorique2506** [comment]
Hello! @Arnav-Singh-5080
I believe i have the necessary skill set and expertise for this issue as i have worked with XGBOOST model before in predicting tennis matches, could u pls assign this to me !
 Thanks!

**Comment by sudeepti25** [comment]
Hi, I have worked with xgboost , I would like to work on this issue > Can you please assign it to me

**Comment by ujala100** [comment]
Hi @Arnav-Singh-5080

I’m interested in contributing to this project and would love to work on this issue. I have experience working on open-source contributions and would be happy to implement the required changes.

Could you please assign this issue to me? Thanks
